tests/phpunit/integration/Modules/Analytics/AdvancedTrackingTest.php

```diff
@@ -0,0 +1,190 @@
+				$this->assertEmpty( $advanced_tracking->get_event_configurations() );
+			} else {
+				$this->update_plugin_detector( $permutation );
+
+				do_action( 'wp_enqueue_scripts' );
+				$actual_event_configurations = $advanced_tracking->get_event_configurations();
+
+				$this->compare_event_configurations( $actual_event_configurations );
+			}
+		}
+	}
+
+	/**
+	 * Updates the active plugin array in the MockPluginDetector.
+	 *
+	 * @param number $permutation represents what permutation of supported plugins to enable.
+	 */
+	private function update_plugin_detector( $permutation ) {
+		foreach ( $this->supported_plugins as $plugin_name => $plugin_configuration ) {
+			if ( 1 == ( $permutation % 2 ) ) {
+				$this->mock_plugin_detector->add_active_plugin( $plugin_name, $plugin_configuration );
+			} else {
+				$this->mock_plugin_detector->remove_active_plugin( $plugin_name );
+			}
+			$permutation = $permutation >> 1;
+		}
+	}
+
+	/**
+	 * Compares returned list of events to expected events.
+	 *
+	 * @param array $actual_event_configs list of Measurement_Event objects returned by Advanced_Tracking.
+	 */
+	private function compare_event_configurations( $actual_event_configs ) {
+		foreach ( $this->mock_plugin_detector->determine_active_plugins( null ) as $plugin_name => $plugin_config ) {
+			$event_list = $plugin_config['event_config_list'];
+			foreach ( $event_list->get_events() as $expected_event_config ) {
+				$found = false;
+				foreach ( $actual_event_configs as $actual_event_config ) {
+					if ( json_encode( $expected_event_config ) === json_encode( $actual_event_config ) ) {
+						$found = true;
+						break;
+					}
+				}
+				$this->assertTrue( $found );
+			}
+		}
+	}
+
+	/**
+	 * Tests if the expected Javascript code is printed for a given sets of events.
+	 */
+	public function test_injected_code() {
+		$advanced_tracking    = new Advanced_Tracking( $this->mock_plugin_detector );
+		$event_configurations = Measurement_Event_Pipe::encode_measurement_event_list( $advanced_tracking->get_event_configurations() );
+
+		$expected_script = <<<INJECT_SCRIPT
+( function() {
+    var eventConfigurations = {$event_configurations};
+	var config;
+	for ( config of eventConfigurations ) {
+		const thisConfig = config;
+		document.addEventListener( config.on, function( e ) {
+			var el = e.target;
+			var matcher = el.matches || el.webkitMatchesSelector || el.mozMatchesSelector || el.msMatchesSelector || el.oMatchesSelector;
+			if ( matcher && ( matcher.call( el, thisConfig.selector ) || matcher.call( el, thisConfig.selector.concat( ' *' ) ) ) ) {
+				alert( 'Got an event called: '.concat( thisConfig.action ) );
+
+				var params = {};
+				if ( "metadata" in thisConfig && null !== thisConfig.metadata ) {
+					params = thisConfig.metadata( params, el );
+				}
+				params['event_category'] = thisConfig.category;
+				console.log(params);
+
+				gtag( 'event', thisConfig.action, params);
+			}
+		}, true );
+	}
+	}
+)();
+INJECT_SCRIPT;
```

Comment:
We probably don't need to replicate the entire script content here, that would be painful to maintain. Either we should somehow take the base script from the production code class, but if that doesn't really work, we don't really need to test this. The functionality of building the configuration etc is much more substantial.

Reply:
Since the injected script from the production code class is basically a string, It would be difficult to extract base script from it.

Reply:
I have removed the unit test.